### PR TITLE
Fix memory leaks when nua start fails by moving nua cleanups from nua_destroy() to nua_stack_deinit()

### PR DESCRIPTION
**Pull request by azzbcc**
the port leak is producing by https://github.com/freeswitch/sofia-sip/blob/2a2c3ac95b2eb553df4bdf409d10ce71f4d6c0d5/libsofia-sip-ua/nua/nua.c#L156

all I need to do is call https://github.com/freeswitch/sofia-sip/blob/2a2c3ac95b2eb553df4bdf409d10ce71f4d6c0d5/libsofia-sip-ua/nua/nua.c#L231

### CODE REVIEW

File: libsofia-sip-ua/nua/nua_stack.c [hunk 1]
```diff
@@ -209,8 +209,16 @@ int nua_stack_init(su_root_t *root, nua_t *nua)
 
 void nua_stack_deinit(su_root_t *root, nua_t *nua)
 {
+  nua_handle_t *nh, *nh_next;
+
   enter;
+  for (nh = nua->nua_handles; nh; nh = nh_next) {
```
Comment (andywolk) [inline libsofia-sip-ua/nua/nua_stack.c:215]: nua handles have their own su_home
See https://github.com/freeswitch/sofia-sip/commit/4ff1fb688e0db7e755d313d2f9f774586b1dacfe

Comment (azzbcc) [reply libsofia-sip-ua/nua/nua_stack.c:215]: Tried v1.13.4 but useless, soa leak still.

**Comment by andywolk** [comment]
Thinking we should not touch nua_create() but move some cleanups into nua_stack_deinit()

**Comment by andywolk** [comment]
@azzbcc I pushed a commit to your PR. What do you think?

**Review by mjerris — APPROVED** [review]

**Review by andywolk — APPROVED** [review]

**Pull request merged**